CMake build not correctly using the "${CPUNAME}_SYSTEM" directive

**Describe the bug**
With the current main branch even if the `${CPUNAME}_SYSTEM` variable is correctly set, it is being ignored.  `SIMULATION` works fine though.

**To Reproduce**
Run `make prep` without `SIMULATION` and a targets.cmake file with  `cpu1_SYSTEM` set to something non-empty.
Unexpected error is generated:

    CMake Error at cmake/mission_build.cmake:332 (message):
      Unable to find toolchain file for default


**Expected behavior**
Should generate build tree and build successfully.

**System observed on:**
GSFC MCP750 test build

**Reporter Info**
Joseph Hickey, Vantage Systems, Inc.

--- a/cmake/global_functions.cmake
+++ b/cmake/global_functions.cmake
@@ -185,10 +185,10 @@ function(read_targetconfig)
   endwhile()
   
   foreach(CPUNAME ${MISSION_CPUNAMES})
-    if (DEFINED SIMULATION)
+    if (SIMULATION)
       # if simulation use simulation system architecture for all targets
       set(TOOLCHAIN_NAME "${SIMULATION}")
-    elseif (DEFINED ${CPUNAME}_SYSTEM)
+    elseif (${CPUNAME}_SYSTEM)
       # get the target system arch identifier string
       set(TOOLCHAIN_NAME "${${CPUNAME}_SYSTEM}")
     else()

